### PR TITLE
#372 Error Strategy on KafkaListener annotation should offer consistency or availability and not skip records in current poll on error

### PR DESCRIPTION
**Pull request by cwebbtw**
As per https://github.com/micronaut-projects/micronaut-kafka/issues/372

### CODE REVIEW

File: kafka/src/main/java/io/micronaut/configuration/kafka/annotation/ErrorStrategy.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * as well as set the error strategy.
+ *
+ * @author Christopher Webb
+ * @author Vishal Sulibhavi
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ErrorStrategy {
+    /**
+     * The delay used with STOP_ON_ERROR {@link io.micronaut.configuration.kafka.annotation.ErrorStrategyValue}
```
Comment (rsolci) [inline kafka/src/main/java/io/micronaut/configuration/kafka/annotation/ErrorStrategy.java:23]: Should't it be `RETRY_ON_ERROR` instead of `STOP_ON_ERROR`?

**Review by rsolci — COMMENTED** [review]
My company uses micronaut and we are using and following this project. Would adding the capability of retrying for a number of times too much work? Because retrying forever may not be suitable for some cases.
Right now we implemented a custom ExceptionHandler that keeps a in memory map of how many times an offset was tried on a given topic and partition

**Comment by cwebbtw** [comment]
> @rsolci My company uses micronaut and we are using and following this project. Would adding the capability of retrying for a number of times too much work? Because retrying forever may not be suitable for some cases.
Right now we implemented a custom ExceptionHandler that keeps a in memory map of how many times an offset was tried on a given topic and partition

This shouldn't be too much extra to add, will take a look - this has taken slightly lower priority following some internal organisation activity that has taken precedence, we should be back on this soon.

**Comment by cwebbtw** [comment]
This was merged into another micronaut branch and is now part of micronaut 4.x

**Pull request closed without merging**